src/cljs/owlet_ui/events/contentful.cljs

```diff
@@ -70,24 +69,41 @@
             :activity-titles activity-titles)
        :dispatch [route-dispatch route-param]})))
 
-;TODO: add :show-skill, :show-activity
+
+; route dispatches
 
 (rf/reg-event-fx
   :show-branches
   (fn [_ _]
-    {:dispatch [:set-active-view :branches-view]}))
+    {:dispatch-n (list [:set-active-view :branches-view]
+                       [:set-active-document-title! "Branches"])}))
+
+(rf/reg-event-fx
+  :show-branch
+  (fn [_ [_ route-param]]
+    {:dispatch-n (list [:set-active-view :filtered-activities-view "branch"]
+                       [:filter-activities-by-search-term route-param]
+                       [:set-active-document-title! route-param])}))
 
 (rf/reg-event-fx
   :show-platform
-  (fn [_ route-param]
+  (fn [_ [_ route-param]]
     {:dispatch-n (list [:set-active-view :filtered-activities-view "platform"]
-                       [:filter-activities-by-search-term (second route-param)])}))
+                       [:filter-activities-by-search-term route-param]
+                       [:set-active-document-title! route-param])}))
 
 (rf/reg-event-fx
-  :show-branch
-  (fn [_ route-param]
+  :show-skill
+  (fn [_ [_ route-param]]
     {:dispatch-n (list [:set-active-view :filtered-activities-view "branch"]
-                       [:filter-activities-by-search-term (second route-param)])}))
+                       [:filter-activities-by-search-term route-param]
+                       [:set-active-document-title! route-param])}))
+
+(rf/reg-event-fx
+  :show-activity
+  (fn [_ [_ route-param]]
+    {:dispatch-n (list [:set-active-view :activity-view]
+                       [:set-activity-in-view route-param])}))
```

Comment:
Definitely heading in the right direction! You can see another example of this event-callback technique in my `:auth0-authenticated` effect, which gets fired by a call back invoked by the Auth0Lock JS object. It in turn invokes effect `:firebase-sign-in` with an argument `:firebase-sign-in-failed`. That event handler is in turn invoked if Auth0 couldn't sign into Firebase. So everything proceeds in order: function --> effect --> effect --> event!

I should mention that in your `:get-content-from-contentful` handler, I don't think you're obtaining the "params" arguments. So `route-param` in the `let` will always be nil. You need an `&` before `route-args` in the arg list.

Re. use of the `:dispatch-n` or `:dispatch` keys: Yes, this makes your handler pure, but the result is the same, doing a bunch of dispatches that could be impure. If the handler you're dispatching to is pure, then don't dispatch. Just call it as a function. We need to be rewriting them as follows:
```
(rf/reg-event-db
  :set-activity-in-view
  (fn [db [_ activity-id all-activities]]
    ... ))

;; BECOMES

(defn set-activity-in-view
  [db [_ activity-id all-activities]]
  ... )
(rf/reg-event-db :set-activity-in-view set-activity-in-view)
```
If, on the other hand, the dispatched-to event handler is impure, then it should be rewritten as an _effect_ handler, i.e., in a call to `reg-fx`. In rare cases like `:set-active-view`, the DOM-manipulating effect probably belongs in a view instead. The DOM-accessing code could be rewritten as JavaScript in a `<script>` tag. The tag, along with the rest of the page, would be built by our view function. That way, we just render the page as normal and the effect stays out there in the impure DOM world, not in our handlers.

Think of events (along with coeffects) as the way _the outside world_ talks to our app, not the way the app talks to itself. Our app talks to the outside world through views and effects. Everything else should be pure!